clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -500,14 +500,24 @@ boolean joinGroupIfNeeded(final Timer timer) {
                     requestRejoin(shortReason, fullReason);
                 }
 
+                // 4 special non-retriable exceptions that we want to retry, as long as the timer hasn't expired.
```

Comment:
I think here the comment is not to just state what the code did, since readers can just understand that from the code :P instead what we want to emphasize is to remind future contributors that they should be careful to not change the precedence ordering of this logic unnecessarily.